src/App.tsx

```diff
@@ -17,18 +85,21 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={setStatus}
+                onSearch={setSearch}
+                onClear={() => setSearch('')}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            <div className="block">{getTodosListView()}</div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal todos={selectedTodo} onClose={() => setSelectedTodo(null)} />
```

Comment:
The `TodoModal` component is being passed a prop named `todos`, but it seems like it should be `todo` or `selectedTodo` given the context. Verify the prop name expected by `TodoModal` and adjust accordingly.